apps/dashboard/src/components/workflow-editor/test-workflow/test-workflow-form.tsx

```diff
@@ -0,0 +1,205 @@
+import { useMemo, useState } from 'react';
+import { RiSendPlaneFill } from 'react-icons/ri';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { Editor } from '@monaco-editor/react';
+import type { WorkflowResponseDto } from '@novu/shared';
+import { Code2 } from '../../icons/code-2';
+import { Panel, PanelContent, PanelHeader } from '../../primitives/panel';
+import { FormControl, FormField, FormItem, FormLabel, FormMessage } from '../../primitives/form/form';
+import { Input, InputField } from '../../primitives/input';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '../../primitives/tabs';
+import { capitalize } from '@/utils/string';
+import {
+  type CodeSnippet,
+  createCurlSnippet,
+  createFrameworkSnippet,
+  createGoSnippet,
+  createNodeJsSnippet,
+  createPhpSnippet,
+  createPythonSnippet,
+} from '@/utils/code-snippets';
+import { CopyButton } from '../../primitives/copy-button';
+import { TestWorkflowFormType } from '../schema';
+
+const tabsTriggerClassName = 'pt-1';
+const codePanelClassName = 'bg-background h-full w-full rounded-lg border border-neutral-200 p-3';
+
+type SnippetLanguage = 'shell' | 'framework' | 'typescript' | 'php' | 'go' | 'python';
+
+const LANGUAGE_TO_SNIPPET_UTIL: Record<SnippetLanguage, (props: CodeSnippet) => string> = {
+  shell: createCurlSnippet,
+  framework: createFrameworkSnippet,
+  typescript: createNodeJsSnippet,
+  php: createPhpSnippet,
+  go: createGoSnippet,
+  python: createPythonSnippet,
+};
+
+const SnippetEditor = ({ language, value }: { language: SnippetLanguage; value: string }) => {
+  const editorLanguage = language === 'framework' ? 'typescript' : language;
+
+  return (
+    <Editor
+      defaultLanguage={editorLanguage}
+      language={editorLanguage}
+      className="h-full"
+      options={{
+        minimap: {
+          enabled: false,
+        },
+        // workaround from: https://github.com/microsoft/monaco-editor/issues/2093
+        accessibilitySupport: 'off',
+        renderLineHighlight: 'none',
+        scrollBeyondLastLine: false,
+        fontSize: 14,
+        lineHeight: 20,
+        readOnly: true,
+      }}
+      value={value}
+    />
+  );
+};
+
+export const TestWorkflowForm = ({ workflow }: { workflow?: WorkflowResponseDto }) => {
+  const {
+    control,
+    formState: { errors },
+  } = useFormContext<TestWorkflowFormType>();
+  const [activeSnippetTab, setActiveSnippetTab] = useState<SnippetLanguage>('framework');
+  const to = useWatch({ name: 'to', control });
+  const payload = useWatch({ name: 'payload', control });
+  const identifier = workflow?.workflowId ?? '';
+  const snippetValue = useMemo(() => {
+    const snippetUtil = LANGUAGE_TO_SNIPPET_UTIL[activeSnippetTab];
+    return snippetUtil({ identifier, to, payload });
+  }, [activeSnippetTab, identifier, to, payload]);
```

Comment:
generate snippet value based on: language, to and payload fields; so it shows live changes in snippets